Rakefile

```diff
@@ -42,3 +43,42 @@ task default: :spec
 
 YARD::Rake::YardocTask.new do |t|
 end
+
+# Executes a command locally.
+#
+# @param command [String] command to execute.
+# @return [Object] the standard out stream.
+def run_local_command(command)
+  stdout, stderr, status = Open3.capture3(command)
+  error_message = "Attempted to run\ncommand:'#{command}'\nstdout:#{stdout}\nstderr:#{stderr}"
+  raise error_message unless status.to_i.zero?
+
+  stdout
+end
+
+# Build the gem
+desc 'Build the gem'
+task :build do
+  gemspec_path = File.join(Dir.pwd, 'ruby-pwsh.gemspec')
+  run_local_command("bundle exec gem build '#{gemspec_path}'")
+end
+
+# Tag the repo with a version in preparation for the release
+#
+# @param :version [String] a semantic version to tag the code with
+# @param :sha [String] the sha at which to apply the version tag
+desc 'Tag the repo with a version in preparation for release'
+task :tag, [:version, :sha] do |_task, args|
+  run_local_command('git fetch upstream')
+  run_local_command("git tag -a version -m #{args[:version]} #{args[:sha]}")
```

Comment:
Probably you should include a validation for args[:version] at least, maybe a regex like `(\d+\.)?(\d+\.)?(\*|\d+)$`

Reply:
Theoretically you could tag with anything, but yeah, happy to do so! 🙂